binary-helper/src/config.rs

```diff
@@ -0,0 +1,82 @@
+use common::config::{DatabaseConfig, GrpcConfig, LoggingConfig, NatsConfig};
+
+use super::Config;
+
+#[derive(Debug, Clone, PartialEq, config::Config, serde::Deserialize)]
+#[serde(default)]
+pub struct AppConfig<T: ConfigExtention> {
+    /// The name of the application
+    pub name: String,
+
+    /// The path to the config file
+    pub config_file: Option<String>,
+
+    /// The logging configuration
+    pub logging: LoggingConfig,
+
+    /// The gRPC configuration
+    pub grpc: GrpcConfig,
+
+    /// The database configuration
+    pub database: DatabaseConfig,
+
+    /// The NATS configuration
+    pub nats: NatsConfig,
+
+    #[serde(flatten)]
+    #[config(flatten)]
+    pub extra: T,
+}
+
+pub trait ConfigExtention: config::Config + Default {
+    const APP_NAME: &'static str;
+
+    fn config_default() -> AppConfig<Self>
+    where
+        Self: config::Config + Default,
+    {
+        AppConfig {
+            name: Self::APP_NAME.to_owned(),
+            config_file: Some("config".to_owned()),
+            logging: Default::default(),
+            grpc: Default::default(),
+            database: Default::default(),
+            nats: Default::default(),
+            extra: Self::default(),
+        }
+    }
```

Comment:
Why is this a function on the `ConfigExtension` trait instead of an `Default` impl for `AppConfig`

Reply:
Rust has namespace rules about trait implementation. AppConfig is in a different crate so i cannot implement traits on that struct.

Reply:
Isn't AppConfig in this same file?

Reply:
No, only an alias.